spec/controllers/pages_controller_spec.rb

```diff
@@ -13,13 +13,36 @@
 
   describe 'GET #secret_page' do
     context 'with correct params' do
-      it 'renders the secret_page template'
-      it 'does not 302 temp redirect'
+
+      let(:user) { create(:user) }
+
+      before(:each) do
+        allow(controller).to receive(:current_user).and_return(user)
+      end
+
+
+      it 'responds with a 200 OK HTTP response' do
+        get :secret_page
+        expect(response.status).to eq(200)
+      end
+
+
+      it 'renders the secret_page template' do
+        get :secret_page
+        expect(response).to render_template "secret_page"
+      end
     end
 
     context 'with incorrect params' do
-      it 'responds with a 302 temp redirect'
-      it 'displays the login page'
+      it 'responds with a 302 temp redirect' do
+        get :secret_page
+        expect(response.status).to eq(302)
+      end
+
+      it 'displays the login page' do
+        get :secret_page
+        expect(response).to redirect_to "/login"
```

Comment:
Could use path variables here again. If you're unsure what the variables are, then got to this endpoint when running locally and it will list them for you http://localhost:3000/rails/info/routes